state/application.go

```diff
@@ -1949,64 +1949,46 @@ func (a *Application) SetDownloadedIDAndHash(id, hash string) error {
 	return nil
 }
 
-func checkSeriesForSetCharm(currentPlatform *Platform, charm *Charm, ForceBase bool) error {
+func checkSeriesForSetCharm(currentPlatform *Platform, ch *Charm, ForceBase bool) error {
 	// For old style charms written for only one series, we still retain
 	// this check. Newer charms written for multi-series have a URL
 	// with series = "".
-	curSeries, err := corebase.GetSeriesFromChannel(currentPlatform.OS, currentPlatform.Channel)
+	curBase, err := corebase.ParseBase(currentPlatform.OS, currentPlatform.Channel)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	charmSeries, err := corecharm.ComputedSeries(charm)
+	charmBases, err := corecharm.ComputedBases(ch)
 	if err != nil {
 		return errors.Trace(err)
 	}
-	if charm.URL().Series != "" {
-		// Allow series change when switching to charmhub charms.
-		// Account for legacy charms with "kubernetes" series in the URL.
```

Comment:
Did you test this type of charm? The old style k8s ones?

Reply:
I didn't think we supported those anymore?

Reply:
Unless we specifically have an error message and tell people they are not supported while giving them an upgrade path, we have to handle it. There are at least 2 filed bugs and a long discussion in the juju MM channel today.